longTrip.py

```diff
@@ -0,0 +1,121 @@
+import pandas as pd
+import math
+
+class Trip:
+    """
+    トリップ情報を保持するクラス
+    
+    Attributes
+    ----------
+    userId: int
+        ユーザID
+    points: dict
+        キーに座標のタイムスタンプ、値にXY座標
+    
+    """
+    
+    userId = -1
+    points = {}
+    
+    def __init__(self, userId):
+        """
+        初期化
+        
+        Attributes
+        ----------
+        userId: int
+            ユーザID
+        """
+        
+        self.userId = userId
+        self.points = {}
+    
+    def addPoint(self, timeStamp, pointX, pointY):
+        """
+        座標の追加
+        
+        Attributes
+        ----------
+        timeStamp: int
+            UNIXタイムでの時刻
+        pointX: int
+            X座標
+        pointY: int
+            Y座標
+        """
+        self.points[timeStamp] = (pointX, pointY)
+        
+    def print(self):
+        print("userId: " + str(self.userId))
+        print(self.points)
+        
+    def calcDistance(self):
+        """
+        当該トリップの移動距離算出
+        pointsデータを参照し、スタート地点からゴール地点までの距離を算出
+        """
+        distance = 0
+        lastX = None
+        lastY = None
+        for point in self.points:
+            curX = self.points[point][0]
+            curY = self.points[point][1]
+            if lastX != None:
+                curDistance = math.sqrt((curX - lastX)**2 + (curY - lastY)**2)
+                distance = distance + curDistance
+            lastX = curX
+            lastY = curY
+        return distance
+
+def calcTop3(trips):
+    distance1 = -1
+    distance2 = -1
+    distance3 = -1
+    id1 = -1
+    id2 = -1
+    id3 = -1
+
+    for trip in trips:
+        if distance1 < trip.calcDistance():
+            distance1 = trip.calcDistance()
+            id1 = trip.userId
+            
+    for trip in trips:
+        if distance2 < trip.calcDistance() and trip.calcDistance() < distance1:
+            distance2 = trip.calcDistance()
+            id2 = trip.userId
+
+    for trip in trips:
+        if distance3 < trip.calcDistance() and trip.calcDistance() < distance2:
+            distance3 = trip.calcDistance()
+            id3 = trip.userId
+    top3 = [[id1, distance1],[id2, distance2],[id3, distance3]]
+    return top3
+
+
+# ここからメイン処理
+df = pd.read_csv("data.csv")
+
+trips = []
+lastUserId = -1
+curTrip = None
+
+# dataを一行ずつ読み込んでTripオブジェクト群を作成
+for row in df.sort_values(by=["userId", "timeStamp"]).itertuples():
+    if lastUserId != row[1]:
+        if curTrip != None:
+            trips.append(curTrip)
+        curTrip = Trip(row[1])
+        lastUserId = row[1]
+    curTrip.addPoint(row[2], row[3], row[4])
+if curTrip != None:
```

Comment:
pep8に従い、==ではなく、isを使用しましょう。


```suggestion
if curTrip is not None:
```